pkg/callback/doc.go

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 ForgeRock AS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package callback provides types for handling authentication callbacks. Use the predefined callback handlers to
+// process callbacks received from Access Management's authentication framework or use the Handler interface to
+// provide your own handler.
+//
+//
+// This is an example of how to create your own callback handler:
+//
+//    type ThingHandler struct {
```

Comment:
Don't need to do this now but this could be moved to an example file e.g. `example_custom.go`. GoDoc would then add it to the docs as a runnable example (I am a bit hazy on the details).

Reply:
Yes I tried this, but it ended up running it as part of the tests, so just adding snippets of code didn't work. Need some more time to play around with it.